src/Compilers/CSharp/Test/Emit2/Semantics/ExperimentalAttributeTests.cs

```diff
@@ -77,18 +80,779 @@ public class C
         var src = """
 C.M();
 """;
+        var comp = CreateCompilation(new[] { src, libSrc, experimentalAttributeSrc });
+        comp.VerifyDiagnostics();
+
+        Assert.Equal(ObsoleteAttributeKind.Experimental, comp.GetTypeByMetadataName("C").ContainingAssembly.ObsoleteKind);
+    }
+
+    [Fact]
+    public void OnAssembly_UsedFromMetadata()
+    {
+        var libSrc = """
+[assembly: System.Diagnostics.CodeAnalysis.Experimental("DiagID1")]
+public class C
+{
+    public static void M() { }
+}
+""";
+
+        var src = """
+C.M();
+""";
+        var comp = CreateCompilation(src, references: new[] { CreateCompilation(new[] { libSrc, experimentalAttributeSrc }).EmitToImageReference() });
+
+        Assert.Equal(ObsoleteAttributeKind.Experimental, comp.GetTypeByMetadataName("C").ContainingAssembly.ObsoleteKind);
+
+        // Note: the assembly-level [Experimental] is equivalent to marking every type and member as experimental,
+        // whereas a type-level [Experimental] is not equivalent to marking every nested type and member as experimental.
```

Comment:
Is there a reason for this? Why not mark only types experimental when assembly is marked experimental? Otherwise members "inherit" obsolete/experimental state from assembly but not from containing type.